src/main/java/com/google/sps/data/MapsRequest.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.data;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.maps.errors.ApiException;
+import com.google.maps.GeoApiContext;
+import com.google.maps.GeoApiContext.Builder;
+import com.google.maps.GeocodingApi;
+import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.LatLng;
+
+/** 
+ * Class that create a GeoApiContext variable and handles requests to maps API.
+ */
+public class MapsRequest {
+  private static GeoApiContext geoApiContext = null;
+
+  /** 
+   * Returns the geoApiContext or creates one if it wasn't already created.
+   */
+  public static GeoApiContext getGeoApiContext() {
+    if (geoApiContext == null) {
+      geoApiContext = new GeoApiContext
+          .Builder()
+          .apiKey("")
```

Comment:
Why empty apikey? Are you planning on setting this up later or is one not required.

Leave a TODO if the former.

Reply:
The apikey is needed, however, I removed it because I received an email from GitGuardian saying "Google Key exposed".

Reply:
Right, not storing them in plain text sounds sensible.

So what are you planning on doing here? Having to go and manually edit random files where this apikey is needed seems complicated. You could have a single `credentials.yaml` file that your applications reads (and it's .gitignored). You could try passing apikeys via environment variables.

You could go fancy and use https://cloud.google.com/secret-manager or https://developer.github.com/v3/actions/secrets/, though I don't know much about what's the easiest way to use those.